### PR TITLE
[solution]: zh/medium-pickbytype

### PR DESCRIPTION
**Pull request by likui628**
# Challenge [zh/medium-pickbytype]

## Checklist

- [x] Make sure that you are opening the PR with a single solution (one solution per PR)
- [x] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

**Comment by likui628** [comment]
```
- ./_site/uk/index.html

  *  External link https://tsch.js.org/610 failed: 404 No error

HTML-Proofer found 1 failure!
```

**Review by ghaiklor — DISMISSED** [review]

**Comment by ghaiklor** [comment]
Did they delete the challenge?

**Comment by ghaiklor** [comment]
Fixed in https://github.com/ghaiklor/type-challenges-solutions/commit/992667043a5464171d6c874e22152537f23f9d2b and https://github.com/ghaiklor/type-challenges-solutions/commit/50cd3383c8e4c772ebb6de947a0f78677ff6e85e

File: zh/medium-pickbytype.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+```typescript
+type PickByType<T, U> = { [P in keyof T]: T[P] };
+```
+
+首先，我们从 `T` 中获得所有的键，并对它们进行迭代。
+在每次迭代时，TypeScript 都会将键赋值给类型 `P`。
+有了键，我们就可以通过查找类型 `T[P]` 来获取值类型。
+
+现在，对迭代应用过滤器将允许我们只找到那些可赋值给 `U` 的。
```
Comment (charleserious) [inline zh/medium-pickbytype.md:41]: May I suggest that we keep the "a" in the translation, so that will be "对迭代应用一个过滤器将允许我们......"。

**Review by charleserious — APPROVED** [review]
LGTM, otherwise

**Review by charleserious — APPROVED** [review]
LGTM

**Pull request merged**